pkg/integrations/v4/supervisor_fb.go

```diff
@@ -144,6 +163,63 @@ func saveToTempFile(config []byte) (string, error) {
 	return file.Name(), nil
 }
 
+func removeFbConfigTempFiles(maxNumberOfFbConfigTempFiles int) ([]string, error) {
+
+	files, err := os.ReadDir(os.TempDir())
+	if err != nil {
+		return nil, err
+	}
+
+	var fbConfigTempFiles []fs.DirEntry
+
+	for _, file := range files {
+		if !file.IsDir() && strings.HasPrefix(file.Name(), "nr_fb_config") {
+			fbConfigTempFiles = append(fbConfigTempFiles, file)
+		}
+	}
+
+	if len(fbConfigTempFiles) > maxNumberOfFbConfigTempFiles {
+
+		var removedConfigTempFilenames []string
+
+		luaFilterTempFileRegex := regexp.MustCompile("nr_fb_lua_filter\\d+")
+
+		sort.Slice(fbConfigTempFiles, func(i, j int) bool {
+			fileInfo1, _ := fbConfigTempFiles[i].Info()
+			fileInfo2, _ := fbConfigTempFiles[j].Info()
+			return fileInfo1.ModTime().Before(fileInfo2.ModTime())
+		})
```

Comment:
That lambda block can not be defined as a constant because it references to fbConfigTempFiles slice and that slice can not be provided as input param. However, the same codo can be written like this (I am not sure if it improves the readability)

```
sortByAscendingDate := func(i, j int) bool {
	fileInfo1, _ := fbConfigTempFiles[i].Info()
	fileInfo2, _ := fbConfigTempFiles[j].Info()
	return fileInfo1.ModTime().Before(fileInfo2.ModTime())
}
		
sort.Slice(fbConfigTempFiles, sortByAscendingDate)
```

Another possibility could be to write a comment just before of that block explaining it does:

```
// sort fbConfigTempFiles by ascending modification date
sort.Slice(fbConfigTempFiles, func(i, j int) bool {
	fileInfo1, _ := fbConfigTempFiles[i].Info()
	fileInfo2, _ := fbConfigTempFiles[j].Info()
	return fileInfo1.ModTime().Before(fileInfo2.ModTime())
})
```